File: Day 1/Variation1.js
function nCr(n, r){ // n represents row number and r represents column number.
    let res = 1;
    for (let i = 0; i<r; i++){
        res = res*(n-i); // This line multiplies the current value of res by n - i. This represents the numerator of the nCr formula (n choose r).
        res = res/(i+1); // We have taken this approach to avoid the ambiguous time complexity.
    }
    return res;
}
function pascalTriangle(r, c){
    const element = nCr(r-1, c-1);
    return element;
}
const r = 5;
const c = 3;
const element = pascalTriangle(r, c);
console.log('Element at position (${r}, ${c}) is: ${element}');

File: Day 1/Variation1.java
import java.util.*;
public class variation1{
public static long nCr(int n, int r){
    long res = 1;
    for (int i = 0; i<r; i++){
        res = res * (n-i);
        res = res/(i+1);
    }
    return res;
}
public static int pascalTriangle(int r, int c){
    int element = (int) nCr(r-1, c-1);
    return element;
}
public static void main(String[] args){
    int r = 6;
    int c = 2;
    int element = pascalTriangle(r, c);
    System.out.println("The element at position (r,c) is: " + element);
}
}